sdk/keyvault/keyvault-keys/src/transformations.ts

```diff
@@ -0,0 +1,55 @@
+import { DeletedKeyBundle, KeyBundle } from "./generated/models";
+import { parseKeyVaultKeyId } from "./identifier";
+import { DeletedKey, KeyVaultKey, JsonWebKey, KeyOperation } from "./keysModels";
+
+/**
+ * @internal
+ * @ignore
+ * Shapes the exposed {@link KeyVaultKey} based on either a received key bundle or deleted key bundle.
+ */
+export function getKeyFromKeyBundle(bundle: KeyBundle | DeletedKeyBundle): KeyVaultKey {
+  const keyBundle = bundle as KeyBundle;
+  const deletedKeyBundle = bundle as DeletedKeyBundle;
+
+  const parsedId = parseKeyVaultKeyId(keyBundle.key!.kid!);
+
+  const attributes: any = keyBundle.attributes || {};
+  delete keyBundle.attributes;
+
+  const resultObject: KeyVaultKey & DeletedKey = {
+    key: keyBundle.key as JsonWebKey,
+    id: keyBundle.key ? keyBundle.key.kid : undefined,
+    name: parsedId.name,
+    keyOperations: keyBundle.key ? (keyBundle.key.keyOps as KeyOperation[]) : undefined,
+    keyType: keyBundle.key ? keyBundle.key.kty : undefined,
+    properties: {
+      expiresOn: attributes.expires,
+      createdOn: attributes.created,
+      updatedOn: attributes.updated,
+      ...keyBundle,
+      ...attributes,
+      ...parsedId,
+      id: keyBundle.key ? keyBundle.key.kid : undefined
+    }
+  };
+
+  if (deletedKeyBundle.deletedDate) {
+    resultObject.properties.deletedOn = deletedKeyBundle.deletedDate;
+    delete (resultObject.properties as any).deletedDate;
+  }
+
+  if (attributes.vaultUrl) {
+    delete (resultObject.properties as any).vaultUrl;
+  }
+  if (attributes.expires) {
+    delete (resultObject.properties as any).expires;
+  }
+  if (attributes.created) {
+    delete (resultObject.properties as any).created;
+  }
+  if (attributes.updated) {
+    delete (resultObject.properties as any).updated;
+  }
```

Comment:
can we find a way to not spread these onto resultObject in the first place? Even if we have to copy attributes to its own object, delete them, then spread the remainder?

Reply:
Oh I would definitely like to do that. I'll do it as part of this PR.

Reply:
I was about to do this, but the context switch is sufficient to me to give me a headache, since this might have consequences on the public API (hopefully not), so I made an issue instead: https://github.com/Azure/azure-sdk-for-js/issues/11720

Reply:
I have an issue to remove all the object spreading on Key Vault: https://github.com/Azure/azure-sdk-for-js/issues/9730 I'll do it as part of that issue!